File: archive/convert.rb
#!/usr/bin/env ruby
require 'sqlite3'
require 'iconv'

rconv = Iconv.new('UTF-8', 'GBK')

db = SQLite3::Database.new('/home/sunway/.tt/pinyin.db')
sql = 'select * from pinyin'
rows = db.execute(sql)
db.execute("delete from pinyin2")
db.execute("begin")
rows.each do |row|
  char = rconv.iconv(row[0])
  db.execute("insert into pinyin2 values (\"#{char}\", \"#{row[1]}\")")
end
db.execute("commit")

sql = 'select * from fuxing'
rows = db.execute(sql)
db.execute("delete from fuxing2")
db.execute("begin")
rows.each do |row|
  char = rconv.iconv(row[0])
  db.execute("insert into fuxing2 values (\"#{char}\", \"#{row[1]}\")")
end
db.execute("commit")

File: bin/gps/convert.py
#!/usr/bin/env python3
import sqlite3
import os
import sys

from .common import open_database

if __name__ == "__main__":
    conn = open_database()
    conn.execute("delete from mud_room_2")
    sql = "select * from mud_room"
    cursor = conn.execute(sql)
    rows = cursor.fetchall()
    for row in rows:
        desc = row[3].replace(" ","")
        tmp = desc.find("你可以看看")
        if tmp != -1:
            desc = desc[:tmp]
        sql = "insert into mud_room_2 values (%d, '%s', '%s', '%s', '%s', '%s', '%s','')" % (row[0],row[1], row[2], desc, row[4],row[5],row[6]);
        conn.execute(sql)
    cursor.close()
    conn.commit()
    conn.close()
    
